### DISCUSSION TITLE
Supported hardware

### DISCUSSION THREAD

**Discussion by wsw70** [krahabb/meross_lan#6] [General]
Just wanted to confirm that the integration correctly recognized single plugs and strips.

The channels in the strips are configured automatically as well:

![image](https://user-images.githubusercontent.com/1345886/111876936-36f99b80-89a1-11eb-9637-e502c98146f3.png)

The only thing that may be a bit misleading is that all the meross devices (their channels, actually) are listed as "related devices":

![image](https://user-images.githubusercontent.com/1345886/111876994-79bb7380-89a1-11eb-8f01-6f03a8c37030.png)

**Comment by krahabb** [comment]
Quoting from https://github.com/krahabb/meross_lan/issues/9#issue-837931909

> If you're interested, I just tested your component with my MSS310H and it works fine.
> This is it:
> ![image](https://user-images.githubusercontent.com/15247405/112029192-43b0f780-8b39-11eb-9bee-1844beef9f13.png)
> 
> I'm writing this, just in case you want to create a list of supported devices 😃

**Comment by Dizzyikea** [comment]
![image](https://user-images.githubusercontent.com/36666298/112054337-58978600-8b4d-11eb-8a4c-6871bf241d34.png)

**Comment by wsw70** [comment]
+ MSS425E

**Comment by wsw70** [comment]
+MSS425F

**Comment by Dizzyikea** [comment]
Just an observation of one of my MSS310's pretty sure my freezer did not spike up to 1500w. Probably a hardware issue not this software but i thought it was interesting to share.
![image](https://user-images.githubusercontent.com/36666298/112117194-fec9a700-8bb2-11eb-8a60-f5059a7d1c2c.png)

**Reply by krahabb** [reply]
Well.. I guess the plug is right. 
It depends on the timings: since the plug sampling rate is not very high they sometime miss the spike which is on average 5 times the nominal power (but could well exceed that depending on wires and so on)
The spike is there for sure when you start an electrical engine with no inverter technology

**Reply by wsw70** [reply]
Yes, the compressor in the refrigerator can easily generate such a spike.

As @krahabb said, the sampling is quite sparse but even with that if you take a running average over a few minutes you would see a pretty much flat line, with some small bumps when it is active (it used to be very dependent on the season but now the fridges are much better)

**Comment by almico** [comment]
@krahabb What kind of devices is supposedly working with your component? RGB bulbs like the MSL120 too?

**Reply by wsw70** [reply]
> MSL120

I am curious how such a device behaves.  Did you connect it to your MQTT and have you seen activity on `/appliance/#`?

**Reply by krahabb** [reply]
Hello,
this component actually works for switches (i.e. plugs and strips) but it could work (just guessing) on bulbs treating them as switches. RGB support is not working for sure.
We could start integrating better support for that if you can provide some hints: we need at least the 'abilities' of the device and/or its response to 'Appliance.System.All'
It is not easy to query the device from the command line since you should create correct checksums and then paste this into a meaningful mqtt message to send to the device (provided it was reconfigured to work on your local mqtt broker)
I'll try to research and guess for a solution with maybe some logging support into my component in order to grab clues about the bulb.
It's definitely something I want to add to this component

Any info about this bulb(s) would be highly appreciated
Thank you!

**Reply by wsw70** [reply]
> RGB support is not working for sure

Is there a specific reason for that? I was hoping that `payload` may carry something different from `togglex` (specifically the brightness, colors etc.) I came to that conclusion since the same app manages the switches and the bulbs (and I may be completely wrong)

**Reply by krahabb** [reply]
No specific reason except I didn't (and still I don't) have any bulb to integrate or test. I'm going to research a bit and start a guessed implementation for that..let's see what happens...
But, if you have any protocol insight for bulbs let me know. Atm I can only start digging into https://github.com/albertogeniola/merossiot

**Reply by wsw70** [reply]
> But, if you have any protocol insight for bulbs let me know

Unfortunately, I just have one smart bulb, on which I flashed ESPhome. I was just wondering whether you knew if there is a specific limitation. 

Of course if I hear about something I will let you know!

**Reply by almico** [reply]
> I just have one smart bulb, on which I flashed ESPhome

@wsw70 do you mean that Meross MSL120 can be flashed with Tasmota? Or were you referring to some other (not Meross) bulb that you flashed with an alternative firmware?

@krahabb I do have a MSL120 I can use for testing. I just need to find a place where to plug it 😄

**Reply by wsw70** [reply]
@almico The latter. I have a generic ESP-based bulb (I do not even remember the name) that I flashed with ESPurna (initially), then moved to ESPHome. I would love to be able to flash Meross devices with ESPHome (or similar: Tasmota, ESPurna) but this is not possible as far as I saw.

**Reply by krahabb** [reply]
@almico I'm going to publish an update to try support light devices...they're a bit more complicated than switches so expect more errors or so...and also some days before releasing

Since you have one of those I'll rely on your beta-testing for sure :)

Just a question right now: did you try to bind the MSL120 to HA-MerossLAN and if so, did it work as a simple switch?
My expectation is that it should work right now, at least switching on/off (my component will see that as a pure switch)

**Reply by almico** [reply]
@krahabb Sorry for the delay. I just opened https://github.com/krahabb/meross_lan/issues/12 😃

**Comment by bytespider** [comment]
Thank you for this. I've just moved one of my MSS310 v2 over and confirm it's working
![homeassistant local_8123_config_devices_device_8730d18b63a2e7511b78de2497fd5201](https://user-images.githubusercontent.com/91801/115063839-021b2e80-9ee4-11eb-95b6-8e776463d520.png)

**Comment by nao-pon** [comment]
@krahabb Thank you so much for this!

I have confirmed that it works perfectly with the following devices.

- mss110 2.0.0 (FW: 2.1.19)
- mss110 2.0.0 (FW: 2.1.20)
- mss110 2.0.0 (FW: 2.1.21)
- mss110 5.0.0 (FW: 5.1.1)
- mss710 3.0.0 (FW: 3.1.6)

**Comment by bobbowitch** [comment]
@krahabb
Thank you for this, found my garagedoor msg100 :) Works with HA automations.
Just got problem integrating with Google Assistant, GA says "This action requires password but no password has been set" when I try and open it with my voice. Not sure where I can put in the PIN/PW in the configuration. 

Device info
msg100 3.5.0
by Meross
Firmware: 3.1.15

**Reply by krahabb** [reply]
Hello @bobbowitch ,
Thank you for confirming the MSG100 works. Related to the google assistant pin I guess it is the security feature included by default in HA so you should be able to manage it in the relevant google assistant configuration of HA. I think it is required by default for 'critical security' devices like locks, doors and garagedoors

**Reply by bobbowitch** [reply]
Hi, 
Added the secure_devices_pin: "xxxx" in configuration.yaml
Works great now. Thanks for the integration!

**Comment by nib2k** [comment]
Hello,

first of all a big thanks goes out to the developer of this fantastic integration.
I have connected my new Meross devices today to my LAN (not Wifi) and the intrgration perfectly recognized them.

It is the Matter series including:
MSH450 : Smart Hub
GS559 : Smart Smoke Detector

Just one question remains from my side: 
What does the "Togglex" Switch do in the Configuration Unit of the smoke detector, any idea or hints?

Thank you and best regards, 
NiB2k

**Reply by krahabb** [reply]
Hello @nib2k,
Thank you for confirming this!
As for the 'togglex', it is a leftover from the fact that every Hub binded device exposes what looks like a general purpose toggle and is consequently exposed by meross_lan.
For some devices (mts1XX valves for example) its use is well defined and effectively turning on/off the device but not for some others.
In meross_lan I have a kind of 'lazy' approach where I leave it exposed by default if I don't know anything about the feature but, whenever, for some devices type, it's clear the feature is useless, the code totally suppresses its creation.
It might be I'm then going to remove it from gs559 since no report so far have ever reported it being useful ;)
